wasmer/config.go

```diff
@@ -166,6 +167,36 @@ func (self *Config) UseCraneliftCompiler() *Config {
 	return self
 }
 
+var opCodeMap map[uint32]uint32 = nil
+
+//export metering_delegate
+func metering_delegate(op C.wasmer_parser_operator_t) C.uint64_t {
+	// a simple alogorithm for now just map from opcode to cost directly
+	// all the responsibility is placed on the caller of PushMiddleware
+	v, b := opCodeMap[uint32(op)]
+	if !b {
+		return 0 // no value means no cost
+	}
+	return C.uint64_t(v)
+}
+
+// PushMiddleware allows the middleware metering to be engaged on a map of opcode to cost
+//   config := NewConfig()
+//	 opmap := map[uint32]uint32{
+//		13: 1,
+//		25: 1,
+//		99: 4,
+//	 }
+//   config.PushMiddleware(7865444, opmap)
+func (self *Config) PushMiddleware(maxGasUsageAllowed uint64, opMap map[uint32]uint32) *Config {
```

Comment:
`PushMeteringMiddleware` is a clearer name (since it's specifically for metering)

Also, it would be a good idea to have a proper enum for the OpCodes instead of int32 directly:

```go
type Season int64

const (
	Unreachable Opcode = 0
	Nop = 1
	Block = 2
)
```

You can get a full list of opcodes here:

https://github.com/bytecodealliance/wasm-tools/blob/main/crates/wasmparser/src/readers/core/operators.rs#L111-L463

Reply:
@syrusakbary Totally makes sense and will do.  I was making sure I was in the correct ballpark first.

Reply:
@syrusakbary Should be all good now.  I tried to verify the codes were all correct.